### ISSUE TITLE
Approch to select tree of inheritance on inheritance_diagram

### ISSUE THREAD

**Issue by vallsv** [sphinx-doc/sphinx#7660]
Hi,

I was playing around inheritance_diagram with the project i am working on (https://github.com/silx-kit/silx/pull/3053). Diagram is very useful so thanks for this project. I would like to know what you think about few points:

1) If there is a link between `A` and `C` between a private class `_B` (and the private flag in not set), the extension do not create a link between `A` and `C`. I think it would be good to draw `A -> C`. If you prefer it could be displayed with a dotted arrow, or with an intermediate anonymous class `A -> [] -> C`. What do you think?  

2) When we specify a top classes, i expect to have only the inheritances (top-down) from that classes. But this extension also generate the whole inheritance of classes inheriting from the top class. I think it is noise, it is not the place to display this relationship.

Example
```
class A: pass
class B (A, Otherthing, Otherthing2): pass
```
With top class = `A`, the extension will also show `Otherthing`, `Otherthing2`. 

3) BTW if i specify a top-class, classes inherited by this class with also be displayed. I don't think it is what you want.

4) A last small thing: It is based on syntactic class name. 2 classes with the same name will be merged together. Is it using python classes internally? Or is it based on text parsing?

To summarize I care about  2 use case for a documentation.

- In a specific class, you want to see how it is implemented, then you want to see the tree of classes inherited (bottom-up)
- In a package, you want to see all the implementations from a class (top-down). (use of `top-classes`?)

But it looks like this extension mix both, and i feel like the result will always be difficult to read, cause you do not ail the same thing. Is there a way to do one or the other of this approaches? Do you think it could be implemented?

**Comment by vallsv** [comment]
Looking at the test code, there is obvious inconsistencies like in `test_diagram_w_1_top_class` (A->B is not shown, while A and B are shown). And in `test_diagram_module_w_2_top_classes` there is comment explaining it is not supposed to work this way.

So i think the initial intention, at least for `top-classes` was what i have proposed. Which is nice :-) Cause it is not a big deal to fix.

I am working on a PR.
